library/Fission/User/Mutation.hs

```diff
@@ -29,14 +29,18 @@ createWithHeroku ::
   -> Text
   -> m (Either Error.Create UserId)
 createWithHeroku herokuUUID herokuRegion username password = runDBNow \now -> do
-  addOnId <- insert HerokuAddOn
-    { herokuAddOnUuid       = herokuUUID
-    , herokuAddOnRegion     = Just herokuRegion
-    , herokuAddOnInsertedAt = now
-    , herokuAddOnModifiedAt = now
-    }
+  let herokuAddOnRecord = HerokuAddOn
+        { herokuAddOnUuid       = herokuUUID
+        , herokuAddOnRegion     = Just herokuRegion
+        , herokuAddOnInsertedAt = now
+        , herokuAddOnModifiedAt = now
+        }
+  insertUnique herokuAddOnRecord >>= \case
+    Just addOnId ->
+      create username password Nothing (Just addOnId) now
 
-  create username password Nothing (Just addOnId) now
+    Nothing ->
+      return (Left Error.AlreadyExists)
```

Comment:
Oh, this is great, though! At the call site, it can capture this is a normal `Left` value and return the `201` 👍

Reply:
Good job on the internal error modelling 🎉

Reply:
Ok so in this case whats the best way to handle their password 😬 ? 

We can't return it. we can create a new one but then technically it isn't idempotent anymore.

Reply:
Yeah, good point. The docs aren't clear so I'll email them 🤷‍♀

Reply:
@expede If you do so can you cc me :)

Reply:
@expede any response?

Reply:
Not yet 🤷‍♀ Whatever let's see if this causes issues